subsys/net/ip/6lo.c

```diff
@@ -46,6 +46,74 @@ static inline u8_t get_6co_cid(struct net_icmpv6_nd_opt_6co *opt)
 static struct net_6lo_context ctx_6co[CONFIG_NET_MAX_6LO_CONTEXTS];
 #endif
 
+static const u8_t udp_nhc_inline_size_table[] = {4, 3, 3, 1};
+
+static int get_udp_nhc_inlined_size(u8_t nhc)
+{
+	int size = 0;
+
+	if ((nhc & 0xF8) != NET_6LO_NHC_UDP_BARE) {
+		NET_DBG("UDP NHC dispatch doesn't match");
+		return 0;
+	}
+
+	if (!(nhc & NET_6LO_NHC_UDP_CHECKSUM)) {
+		size += 2U;
+	}
+
+	size += udp_nhc_inline_size_table[(nhc & NET_6LO_NHC_UDP_PORT_MASK)];
+
+	NET_DBG("Size of inlined UDP HDR data: %d", size);
+
+	return size;
+}
+
+static const u8_t tf_inline_size_table[] = {4, 3, 1, 0};
+/* The first bit of the index is SAC        |  SAC=0   |  SAC=1   |*/
+static const u8_t sa_inline_size_table[] = {16, 8, 2, 0, 0, 8, 2, 0};
+
+static const u8_t da_inline_size_table[] =
+/* The first bit is M, the second DAC
+ *	| M=0 DAC=0 | M=0 DAC=1 | M=1 DAC=0  | M=1 DAC=1 (DAM always 00)
+ */
+	{16, 8, 2, 0, 0, 8, 2, 0, 16, 6, 4, 1, 6};
```

Comment:
Move these variable at the top of the file

Reply:
Done